jsrc/conversions.cpp

```diff
@@ -0,0 +1,637 @@
+/* Copyright 1990-2010, Jsoftware Inc.  All rights reserved.               */
+/* Licensed use only. Any other use is in violation of copyright.          */
+/*                                                                         */
+/* Conversions Amongst Internal Types                                      */
+
+#include <algorithm>
+#include <limits>
+#include <numeric>
+#include <optional>
+
+#include "array.hpp"
+extern "C" {
+#include "verbs/vcomp.h"
+}
+
+#define CVCASE(a, b) (((a) << 3) + (b))  // The main cases fit in low 8 bits of mask
+
+// fuzzy_equal() is used in bcvt, where FUZZ may be set to 0 to ensure only exact values are demoted to lower precision
+// used when v is known to be exact integer. It's close enough, maybe ULP too small on the high end
+[[nodiscard]] static auto
+fuzzy_equal(double u, double v, double fuzz) -> bool {
+    return std::abs(u - v) <= fuzz * std::abs(v);
+}
+
+template <typename T, typename V>
+[[nodiscard]] constexpr auto
+in_range(V value) -> bool {
+    if constexpr (std::is_same_v<bool, V>) {
+        (void)value;
```

Comment:
What is this for?

Reply:
This seems indeed unnecessary, I think I had some compiler errors with bool at the time, but don't remember exactly what

Reply:
Found the reason for the special case, however, I think I'll just disable the warning
```
../jsrc/conversions.cpp:29:45: error: comparison of constant ‘-9223372036854775808’ with boolean expression is always true [-Werror=bool-compare]
--
```